src/main/java/com/github/jmchilton/blend4j/galaxy/JobsClient.java

```diff
@@ -0,0 +1,32 @@
+package com.github.jmchilton.blend4j.galaxy;
+
+import java.util.List;
+
+import com.github.jmchilton.blend4j.galaxy.beans.Job;
+import com.github.jmchilton.blend4j.galaxy.beans.JobDetails;
+
+/**
+ * Client for interacting with Galaxy's Job API.
+ * 
+ * @author Franklin Bristow franklin.bristow@phac-aspc.gc.ca
+ *
+ */
+public interface JobsClient {
+
+	/**
+	 * Get *all* jobs executed by Galaxy. **WARNING**: This may take a long time
+	 * to execute, depending on how many jobs Galaxy has executed in its
+	 * history.
+	 * 
+	 * @return a collection of all jobs executed by Galaxy.
+	 */
+	public List<Job> getJobs();
```

Comment:
You're right, so I will add that feature, because it could be handy.

To clarify, since we're starting with a `HistoryContentsProvenance`, we already have access to the specific job ID, so we wouldn't ever actually get *any* kind of collection of jobs in the place where this is going to be used, we'll access the specific job directly.